common/core/network.go

```diff
@@ -221,6 +226,48 @@ func (plugin *netPlugin) Add(args *cniSkel.CmdArgs) (resultError error) {
 	return nil
 }
 
+func addEndpointGatewaysFromConfig(
+	endpointInfo *network.EndpointInfo,
+	cniConfig *cni.NetworkConfig) {
+
+	defaultDestipv4, defaultDestipv4Network, _ := net.ParseCIDR("0.0.0.0/0")
+	defaultDestipv6, defaultDestipv6Network, _ := net.ParseCIDR("::/0")
+
+	for _, addr := range cniConfig.AdditionalRoutes {
+
+		var isv4 bool
+		if addr.GW.To4() != nil {
+			isv4 = true
+		}
+
+		if isv4 {
+			if endpointInfo.Gateway == nil {
+			   m1, _ := addr.Dst.Mask.Size()
+			   m2, _ := defaultDestipv4Network.Mask.Size()
+
+			   if m1 == m2 &&
+			      addr.Dst.IP.Equal(defaultDestipv4) {
+					endpointInfo.Gateway = addr.GW
+				}
+			}
+		} else {
+				if endpointInfo.Gateway6 == nil {
```

Comment:
nit: looks like this is indented one too many times

Reply:
Fixed.